src/main/java/com/github/manolo8/darkbot/core/objects/facades/SeassonPassMediator.java

```diff
@@ -0,0 +1,136 @@
+package com.github.manolo8.darkbot.core.objects.facades;
+
+import com.github.manolo8.darkbot.core.itf.Updatable;
+import com.github.manolo8.darkbot.core.objects.facades.QuestProxy.Quest;
+import com.github.manolo8.darkbot.core.objects.swf.FlashList;
+
+import eu.darkbot.api.managers.QuestAPI;
+import eu.darkbot.api.managers.SeassonPassAPI;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.ToString;
+
+import static com.github.manolo8.darkbot.Main.API;
+
+import java.util.List;
+
+import org.jetbrains.annotations.Nullable;
+
+public class SeassonPassMediator extends Updatable implements SeassonPassAPI {
+
+    @Getter
+    private CurrentLevelProgress currentLevelProgress = new CurrentLevelProgress();
+
+    @Getter
+    private CurrentLevelProgress currentSeassonLevelInfo = new CurrentLevelProgress();
+
+    @Getter
+    private boolean seassonPassAvailable = false;
+
+    private boolean dailyQuestsUpdated = false;
+    private final FlashList<SeassonPassQuest> dailyQuests = FlashList.ofVector(SeassonPassQuest.class);
+    private boolean weeklyQuestsUpdated = false;
+    private final FlashList<SeassonPassQuest> weeklyQuests = FlashList.ofVector(SeassonPassQuest.class);
+    private boolean seassonQuestsUpdated = false;
+    private final FlashList<SeassonPassQuest> seassonQuests = FlashList.ofVector(SeassonPassQuest.class);
+
+    private long questDataAddr = 0;
+
+    @Override
+    public void update() {
+        if (address == 0) {
+            return;
+        }
+
+        this.dailyQuestsUpdated = false;
+        this.weeklyQuestsUpdated = false;
+        this.seassonQuestsUpdated = false;
+
+        long data = API.readAtom(address + 0x60);
+        this.seassonPassAvailable = API.readBoolean(API.readAtom(data + 0x70) + 0x20);
+
+        if (!seassonPassAvailable) {
+            return;
+        }
+
+        this.currentLevelProgress.update(API.readAtom(data + 0x58));
+        this.currentSeassonLevelInfo.update(API.readAtom(data + 0x68));
+        this.questDataAddr = API.readAtom(address + 0x78);
+    }
+
+    public @Nullable List<? extends SeassonPassQuest> getDailyQuests() {
+        if (!dailyQuestsUpdated && questDataAddr != 0) {
+            dailyQuestsUpdated = true;
+            dailyQuests.update(API.readAtom(questDataAddr + 0x60));
+        }
+
+        return this.dailyQuests;
+    }
+
+    public @Nullable List<? extends SeassonPassQuest> getWeeklyQuests() {
+        if (!weeklyQuestsUpdated && questDataAddr != 0) {
+            weeklyQuestsUpdated = true;
+            weeklyQuests.update(API.readAtom(questDataAddr + 0x68));
+        }
+
+        return this.weeklyQuests;
+    }
+
+    public @Nullable List<? extends SeassonPassQuest> getSeassonQuests() {
+        if (!seassonQuestsUpdated && questDataAddr != 0) {
+            seassonQuestsUpdated = true;
+            seassonQuests.update(API.readAtom(questDataAddr + 0x70));
+        }
+
+        return this.seassonQuests;
+    }

```

Comment:
how big are these lists of quests? is it 100 quests or like 10? if it's a small list (like 5 or 10) there's no need to do this optimization of updating once required

Reply:
From 5 to 20 each list. Then there is a list that is all there that I have not implemented so I could implement that list with this system and you are without this optimisation.